src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/CacheProxy.java

```diff
@@ -0,0 +1,79 @@
+package com.sonyericsson.hudson.plugins.gerrit.trigger.hudsontrigger;
+
+import com.sonyericsson.hudson.plugins.gerrit.trigger.hudsontrigger.data.GerritProject;
+import hudson.util.TimeUnit2;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Keeps map between url and dynamic trigger configuration.
+ * It's used to speed up execution time during updating
+ * trigger jobs and reduce number of connections for the duplicated configs.
+ */
+final class CacheProxy {
```

Comment:
This needs a better name to indicate that it's a proxy for fetching dynamic projects.